src/agent/onefuzz/src/process.rs

```diff
@@ -126,3 +133,81 @@ pub async fn run_cmd<S: ::std::hash::BuildHasher>(
     // convert processcontrol::Output into our Output
     runner.await?.map(|result| result.into())
 }
+
+async fn monitor_stream(name: &str, context: &str, stream: impl AsyncRead + Unpin) -> Result<()> {
+    let mut stream = BufReader::new(stream);
+    loop {
+        let mut buf = vec![];
+
+        let bytes_read = stream.read_until(b'\n', &mut buf).await?;
+        if bytes_read == 0 && buf.is_empty() {
+            break;
+        }
+        let mut line = String::from_utf8_lossy(&buf).to_string();
+        if line.len() > MAX_LOG_LINE_LENGTH {
+            line.truncate(MAX_LOG_LINE_LENGTH);
+            line.push_str("...<truncated>");
+        }
+
+        info!("process ({}) {}: {}", name, context, line);
+    }
+    Ok(())
+}
+
+async fn monitor_process_child(
```

Comment:
The name of this compared to the function below is not very illuminating, mostly because of the generality of the word "monitor". Also, `monitor_process_child()` seems to be the same conceptually as `monitor_process()`, since the `Child` is our handle on the process in question. The real difference is that in `monitor_process()`, we're sipping at the stdio streams, right? Could you add some brief item-level comments about intent, and consider an alternate name for one of them?